operator/controllers/resources/saml_connector_controller.go

```diff
@@ -0,0 +1,71 @@
+package resources
+
+import (
+	"context"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gravitational/teleport/api/types"
+	resourcesv2 "github.com/gravitational/teleport/operator/apis/resources/v2"
+	"github.com/gravitational/teleport/operator/sidecar"
+	"github.com/gravitational/trace"
+)
+
+type SAMLConnectorReconciler struct {
+	*TeleportResourceReconciler[types.SAMLConnector, *resourcesv2.TeleportSAMLConnector]
+	TeleportClientAccessor sidecar.ClientAccessor
+}
+
+func NewSAMLConnectorReconciler(client kclient.Client, accessor sidecar.ClientAccessor) *SAMLConnectorReconciler {
```

Comment:
Is it possible to invert the logic and instead of returning `SAMLConnectorReconciler` we return 
`*TeleportResourceReconciler[...]` . I didn't look too much into this, but my feeling is that we can easily move all the logic into TeleportResourceReconciler and it would simplify  when maintaining the code since the only thing we need to pass is an interface

```go
type testInterface[T types.Resource] interface{
func Get(context.Context, string) (T, error)
func Create(context.Context, T) error
func Update(context.Context, T) error
func Delete(context.Context, string) error
}
```
 and the operator resource.